registry-library/library/library.go

```diff
@@ -0,0 +1,190 @@
+package library
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"text/tabwriter"
+	"time"
+
+	"github.com/containerd/containerd/remotes/docker"
+	"github.com/deislabs/oras/pkg/content"
+	"github.com/deislabs/oras/pkg/oras"
+	indexSchema "github.com/devfile/registry-support/index/generator/schema"
+)
+
+const (
+	// Devfile media types
+	DevfileConfigMediaType  = "application/vnd.devfileio.devfile.config.v2+json"
+	DevfileMediaType        = "application/vnd.devfileio.devfile.layer.v1"
+	DevfileVSXMediaType     = "application/vnd.devfileio.vsx.layer.v1.tar"
+	DevfileSVGLogoMediaType = "image/svg+xml"
+	DevfilePNGLogoMediaType = "image/png"
+	DevfileArchiveMediaType = "application/x-tar"
+
+	scheme                = "http"
+	httpRequestTimeout    = 30 * time.Second // httpRequestTimeout configures timeout of all HTTP requests
+	responseHeaderTimeout = 30 * time.Second // responseHeaderTimeout is the timeout to retrieve the server's response headers
+)
+
+var (
+	stacksPath = os.Getenv("DEVFILE_STACKS")
+	indexPath  = os.Getenv("DEVFILE_INDEX")
+)
+
+// GetRegistryIndex gets the index of the specified registry
+func GetRegistryIndex(registry string) ([]indexSchema.Schema, error) {
+	// Call index server REST API to get the index
+	url := scheme + "://" + path.Join(registry, "index")
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			ResponseHeaderTimeout: responseHeaderTimeout,
+		},
+		Timeout: httpRequestTimeout,
+	}
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	bytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	var registryIndex []indexSchema.Schema
+	err = json.Unmarshal(bytes, &registryIndex)
+	if err != nil {
+		return nil, err
+	}
+	return registryIndex, nil
+}
+
+// ListRegistryStacks lists the stacks of the registry
+func ListRegistryStacks(registry string) error {
+	// Get the registry index
+	registryIndex, err := GetRegistryIndex(registry)
+	if err != nil {
+		return err
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
```

Comment:
I don't think we should directly print out to stdout here. We should leave any printing or formatting to whatever is calling this function, as we're not sure what format they want it displayed in, how it's being displayed, or if there's additional/less information they need displayed.

If we do need this function, we should rename it to `PrintRegistryStacks` and/or return a formatted string rather than printing to stdout.

Reply:
Agree with you, I just create this function for testing the registry list and this function is only used by the CLI, I will add another generic in the middle.